tests/batching_test.py

```diff
@@ -291,6 +291,11 @@ def testDot4(self):
     ans = vmap(np.dot, in_axes=(1, None))(xs, ys)
     expected = onp.einsum('ij,i->j', xs, ys)
     self.assertAllClose(ans, expected, check_dtypes=False)
+  
+  def testDot5(self):
```

Comment:
We have more general tests in https://github.com/google/jax/blob/65f80d6ab23de9436b2b2132b8d0af67938a7ca1/tests/lax_test.py#L2460-L2504

Can you check if those tests exercise this new code path?

Reply:
They do appear to exercise all branches of the new code. Should I also check that no `broadcast` shows up in the jaxpr there?

Reply:
Nah, I just wanted to verify that those had the right coverage. This way we have even more confidence that this change is correct :)